Framework/Muon/src/LoadAndApplyMuonDetectorGrouping.cpp

```diff
@@ -0,0 +1,366 @@
+
+  declareProperty(
+      "ApplyAsymmetryToGroups", true,
+      "Whether to calculate group asymmetry and store the workspaces.");
+
+  // Optional properties
+
+  declareProperty("RebinArgs", emptyString,
+                  "Rebin arguments. No rebinning if left empty.",
+                  Direction::Input);
+
+  declareProperty("TimeOffset", 0.0,
+                  "Shift the times of all data by a fixed amount. The value "
+                  "given corresponds to the bin that will become 0.0 seconds.",
+                  Direction::Input);
+
+  declareProperty("SummedPeriods", std::to_string(1),
+                  "A list of periods to sum in multiperiod data.",
+                  Direction::Input);
+
+  declareProperty("SubtractedPeriods", emptyString,
+                  "A list of periods to subtract in multiperiod data.",
+                  Direction::Input);
+
+  declareProperty(
+      make_unique<WorkspaceProperty<TableWorkspace>>(
+          "DeadTimeTable", "", Direction::Input, PropertyMode::Optional),
+      "Table with dead time information, used to apply dead time correction.");
+
+  // Perform Group Associations.
+
+  std::string analysisGrp("Analysis Options");
+  setPropertyGroup("RebinArgs", analysisGrp);
+  setPropertyGroup("TimeOffset", analysisGrp);
+  setPropertyGroup("SummedPeriods", analysisGrp);
+  setPropertyGroup("SubtractedPeriods", analysisGrp);
+  setPropertyGroup("DeadTimeTable", analysisGrp);
+}
+
+/**
+ * Performs validation of inputs to the algorithm.
+ * - Check the input workspace and WorkspaceGroup are not the same
+ */
+std::map<std::string, std::string>
+LoadAndApplyMuonDetectorGrouping::validateInputs() {
+  std::map<std::string, std::string> errors;
+
+  if (!this->isDefault("WorkspaceGroup")) {
+
+    WorkspaceGroup_sptr groupedWS = getProperty("WorkspaceGroup");
+    Workspace_sptr inputWS = getProperty("InputWorkspace");
+
+    if (groupedWS && groupedWS->getName() == inputWS->getName()) {
+      errors["WorkspaceGroup"] = "The WorkspaceGroup should not have "
+                                 "the same name as InputWorkspace.";
+    }
+  }
+
+  return errors;
+}
+
+void LoadAndApplyMuonDetectorGrouping::exec() {
+  this->setRethrows(true);
+
+  AnalysisOptions options = setDefaultOptions();
+  options.grouping = loadGroupsAndPairs();
+
+  Workspace_sptr inputWS = getProperty("InputWorkspace");
+
+  checkDetectorIDsInWorkspace(options.grouping, inputWS);
+
+  WorkspaceGroup_sptr groupedWS = getProperty("WorkspaceGroup");
+  if (!groupedWS) {
+    groupedWS = addGroupedWSWithDefaultName(inputWS);
+  }
+
+  // Perform the analysis specified in options
+  addGroupingToADS(options, inputWS, groupedWS);
+  addPairingToADS(options, inputWS, groupedWS);
+
+  if (getProperty("ApplyAsymmetryToGroups")) {
+    options.plotType = PlotType::Asymmetry;
+    addGroupingToADS(options, inputWS, groupedWS);
+  }
+  addGroupingInformationToADS(options.grouping);
```

Comment:
We currently do not produce this sort of table. I think we should make it an option of the algorithm if to produce it or not.